library/src/main/java/com/novoda/downloadmanager/LiteDownloadMigrationService.java

```diff
@@ -16,23 +16,30 @@
 
 public class LiteDownloadMigrationService extends Service {
 
+    private static final String TAG = "MigrationService";
     private Migrator v1ToV2Migrator;
     private ExecutorService executor;
     private IBinder binder;
     private Migrator.Callback migrationCallback;
     private String updateMessage;
 
+    private static boolean isBound;
+
+    public static boolean isBound() {
+        return isBound;
+    }
+
     @Override
     public int onStartCommand(Intent intent, int flags, int startId) {
         super.onStartCommand(intent, flags, startId);
 
-        Log.d(getClass().getSimpleName(), "onStartCommand");
+        Log.d(TAG, "onStartCommand");
 
-        if (v1ToV2Migrator != null) {
+        if (v1ToV2Migrator != null && !v1ToV2Migrator.isRunning()) {
```

Comment:
can `v1ToV2Migrator` be null?

`.isRunning()` can be kept private and removed from the interface, you can call the migrator and if its running will just do nothing. But if you use a static executor, as only allows one task at a time, you don't even need to check for this.

Reply:
No, it can't be `null` really 😄  I'll remove both of these and go for the `static` executor.